adagucserverEC/CAutoConfigure.cpp

```diff
@@ -484,10 +484,11 @@ int CAutoConfigure::justLoadAFileHeader(CDataSource *dataSource) {
     return 0;
   }
 
-  CT::string foundFileName;
-
-  /* Use the file specified as header file */
-  foundFileName = dataSource->headerFileName.c_str();
+  CT::string foundFileName = dataSource->getFileName();
+  if (foundFileName.empty()) {
+    /* Use the file specified as header file */
+    foundFileName = dataSource->headerFileName.c_str();
+  }
```

Comment:
It is not clear to me what the difference between `fileName` and `headerFileName` is, and how this is affected by animationStep.

Reply:
- `dataSource->getFileName()` has the filename for the current requested set of dimension. This is used when reading actual data. But it is sometimes not yet present, as in certain cases the header information needs to be read first.
- `dataSource->headerFileName()` points to one of the files in the sequence of files for a layer. This can be used to read metadata.

justLoadAFileHeader tries to load netcdf header information. When there is already a file name available in headerFileName, that can be used. It will prevent costly directory scans or database lookups.